supporting-blog-content/context-engineering-mistral-completions/notebook.ipynb

```diff
@@ -0,0 +1,517 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "fbc1cebe",
+   "metadata": {},
+   "source": [
+    "# Mistral Chat Completions with Elasticsearch Inference API\n",
+    "\n",
+    "This notebook demonstrates how to set up a Mistral chat completion inference endpoint in Elasticsearch and stream chat responses using the inference API\n",
+    "\n",
+    "## Prerequisites\n",
+    "- Elasticsearch cluster \n",
+    "- Elasticsearch API key\n",
+    "- Mistral API key"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7a7f9f82",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install requests tqdm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f284f828",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import json\n",
+    "import os\n",
+    "from typing import Dict, Any, Generator\n",
+    "from IPython.display import display, Markdown, clear_output\n",
+    "import time\n",
+    "\n",
+    "# set env PYTHONUNBUFFERED=1\n",
+    "os.environ[\"PYTHONUNBUFFERED\"] = \"1\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "196bb0a3",
+   "metadata": {},
+   "source": [
+    "## Configuration\n",
+    "\n",
+    "Set up your Elasticsearch and Mistral API credentials. For security, consider using environment variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5fdf1b27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Configuration - Update these values with your credentials\n",
+    "ELASTICSEARCH_HOST = \"https://tralmis-tral-d91b08.es.us-central1.gcp.elastic.cloud:443\"\n",
+    "ELASTICSEARCH_API_KEY = \"elasticsearchapikey==\"\n",
+    "MISTRAL_API_KEY = \"mistralapikey==\"  # Replace with your Mistral API key\n",
+    "\n",
+    "MISTRAL_MODEL = \"mistral-large-latest\"  # Mistral model to use\n",
+    "INFERENCE_ENDPOINT_NAME = (\n",
+    "    \"mistral-embeddings-chat-completion\"  # Name for the inference endpoint\n",
+    ")\n",
+    "\n",
+    "ELASTICSEARCH_HEADERS = {\n",
+    "    \"Authorization\": f\"ApiKey {ELASTICSEARCH_API_KEY}\",\n",
+    "    \"Content-Type\": \"application/json\",\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "20d4ceda",
+   "metadata": {},
+   "source": [
+    "## Create the Inference Endpoint\n",
+    "\n",
+    "Create the Mistral chat completion endpoint if it doesn't exist."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bdbdfaef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Couldn't we use the Elastic Python Client to create this?

https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-inference-put-mistral